WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewController.swift

```diff
@@ -129,16 +158,37 @@ private extension ShippingLabelFormViewController {
-                                                                           type: .destination,
-                                                                           address: shippingLabelAddress,
-                                                                           completion: { [weak self] (shippingLabelAddress) in
-                                                                            guard let self = self else { return }
-                                                                            self.viewModel.handleDestinationAddressValueChanges(address: shippingLabelAddress)
-                                                                           })
-            self.navigationController?.pushViewController(shippingAddressVC, animated: true)
+
+            self.viewModel.validateAddress(type: .destination) { [weak self] (validationState, response) in
+                guard let self = self else { return }
+                let shippingLabelAddress = self.viewModel.destinationAddress
+                switch validationState {
+                case .validated:
+                    self.viewModel.handleDestinationAddressValueChanges(address: shippingLabelAddress,
+                                                                   validated: true)
+                case .suggestedAddress:
+                    let vc = ShippingLabelSuggestedAddressViewController(type: .destination,
+                                                                         address: shippingLabelAddress,
+                                                                         suggestedAddress: response?.address) { [weak self] (newShippingLabelAddress) in
+                        self?.viewModel.handleDestinationAddressValueChanges(address: newShippingLabelAddress,
+                                                                        validated: true)
+                    }
+                    self.navigationController?.pushViewController(vc, animated: true)
+                default:
+                    let shippingAddressVC = ShippingLabelAddressFormViewController(siteID: self.viewModel.siteID,
+                                                                                   type: .destination,
+                                                                                   address: shippingLabelAddress,
+                                                                                   completion: { [weak self] (newShippingLabelAddress) in
+                                                                                    guard let self = self else { return }
+                            self.viewModel.handleDestinationAddressValueChanges(address: newShippingLabelAddress, validated: true)
+
+                                                                                   })
+                    self.navigationController?.pushViewController(shippingAddressVC, animated: true)
+                }
```

Comment:
nit: decouple VC navigation like in `configureShipFrom` suggestion above, maybe reuse same methods if that will be clean.

Reply:
✅  a79b10b3c29233f5525a598f65bcc3d84aab081c